### PR TITLE
Added beacon name sync

### PR DESCRIPTION
**Pull request by Measurity**
- Added beacon sync
- Added beacon persist to server for new players or reconnecting players
- Fixes a locking issue
- Fixed equipment helper checking for null on component types instead of doing it 'the unity way'
- Removed unused GenericNode class
- Fixed code styling for array initializers
- Refactored and added defensive checks to player stats UI so that it doesn't crash if elements go missing (due to gamemode change)

Suppressed the ping/beacon rename packet on initial sync so that connecting players don't fire beacon rename packets.

Initially I used `PingManager.onRenamed` event but doing it this way would deviate from the standard patching mechanism so I went with a patch instead.

### CODE REVIEW

**Comment by Measurity** [comment]
Bump

File: NitroxClient/Communication/Packets/Processors/InitialPlayerSyncProcessor.cs [hunk 1]
```diff
@@ -36,19 +39,21 @@ public override void Process(InitialPlayerSync packet)
         
         private IEnumerator ProcessInitialSyncPacket(object sender, EventArgs eventArgs)
         {
-            bool moreProcessorsToRun;
-
-            do
+            // Some packets should not fire during game session join but only afterwards so that initialized/spawned game objects don't trigger packet sending again. 
+            using (packetSender.Suppress<PingRenamed>())
```
Comment (Sunrunner37) [inline NitroxClient/Communication/Packets/Processors/InitialPlayerSyncProcessor.cs:43]: Should we add this closer to the code that invokes this?

File: NitroxClient/Communication/Packets/Processors/PingRenamedProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            if (!beacon)
+            {
+                // This can be ok if origin of ping instance component was not from a beacon (but from signal or other).
+                return;
+            }
+            if (beacon.GetComponent<Player>())
+            {
+                // Skip over beacon component on player GameObjects
+                return;
+            }
```
Comment (Sunrunner37) [inline NitroxClient/Communication/Packets/Processors/PingRenamedProcessor.cs:42]: log the ones that are error?

File: NitroxClient/GameLogic/ChatUI/PlayerChat.cs [hunk 3]
```diff
@@ -60,8 +60,12 @@ public void HideLog()
 
         public void ShowChat()
         {
-            ShowLog();
+            if (!InputField.CanEnable())
```
Comment (Sunrunner37) [inline NitroxClient/GameLogic/ChatUI/PlayerChat.cs:63]: What does this check do?

File: NitroxClient/Unity/Helper/GameObjectHelper.cs [hunk 4]
```diff
@@ -55,5 +57,45 @@ public static GameObject RequireGameObject(string name)
...
         }
+
+        /// <summary>
+        ///     Calls the selector if the Unity object is alive. Returns null without call if not alive.
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <param name="selector"></param>
+        /// <typeparam name="TIn"></typeparam>
+        /// <typeparam name="TOut"></typeparam>
+        /// <returns></returns>
```
Comment (Sunrunner37) [inline NitroxClient/Unity/Helper/GameObjectHelper.cs:68]: These are not providing value.  Same below.

File: NitroxServer/Player.cs [hunk 5]
```diff
@@ -137,15 +137,13 @@ public void SendPacket(Packet packet)
...
+            if (obj == null)
             {
                 return false;
             }
 
-            Player player = (Player)obj;
-
-            return player.Id == Id;
+            Player player = obj as Player;
+            return player != null && player.Id == Id;
```
Comment (Sunrunner37) [inline NitroxServer/Player.cs:146]: We already checked for null above.

File: NitroxModel/Packets/PingRenamed.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    public class PingRenamed : Packet
+    {
+        [ProtoMember(1)]
+        public NitroxId Id { get; }
+
+        [ProtoMember(2)]
+        public string Name { get; }
+        
+        [ProtoMember(3)]
+        public byte[] BeaconGameObjectSerialized { get; }
```
Comment (Sunrunner37) [inline NitroxModel/Packets/PingRenamed.cs:18]: Do all pings have beacons?

Comment (Measurity) [reply NitroxModel/Packets/PingRenamed.cs:18]: No, all beacons have pings.

File: NitroxClient/Communication/Packets/Processors/PingRenamedProcessor.cs (hunk 2 above)
Comment (Measurity) [reply NitroxClient/Communication/Packets/Processors/PingRenamedProcessor.cs:42]: None of these have to be errors. And this information doesn't seem useful most of the time. I wish we had some `.Trace()` or just disable logs from this class by default.

**Pull request merged**